labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise01.scala

```diff
@@ -99,7 +98,9 @@ object ListManipulationExercise01 {
    * Implement it whatever way suites you best. Hint: it can be done in a neat way using recursion. 
    */
   def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+    // base case
+    if (l.isEmpty) List(List())
```

Comment:
I'm kind of surprised that this line works. Does Scala just figure out what the element type of the empty list should be, or does the empty list not have an element type?